tests/test_projectbuilder.py

```diff
@@ -573,71 +579,51 @@ def test_log(mocker, caplog, test_flit_path):
         assert caplog.records[-1].lineno == 562
 
 
-def test_parse_build_system_table_type_validation():
-    with pytest.raises(
-        build.BuildSystemTableValidationError,
-        match='`requires` is a required property',
-    ):
-        build._parse_build_system_table(
+@pytest.mark.parametrize(
+    ['pyproject_toml', 'raises'],
+    [
+        (
             {'build-system': {}},
-        )
-
-    with pytest.raises(
-        build.BuildSystemTableValidationError,
-        match='`requires` must be an array of strings',
-    ):
-        build._parse_build_system_table(
+            pytest.raises(build.BuildSystemTableValidationError, match='`requires` is a required property'),
```

Comment:
We can just parameterize the match.

https://github.com/FFY00/python-pep621/blob/f374fa2fe8f53fc3b24ef9e08b00870fef8fce30/tests/test_standard_metadata.py#L479

Reply:
Not all of them raise.

Reply:
Then split the test into one for the errors and one for the case it should pass.

Reply:
Do you not like the way it's structured now or why do you want me to do that?

Reply:
It's to remove duplication and complexity. If you split the test into two you can easily parameterize only the error string.

Reply:
I hope https://github.com/pypa/build/pull/365/commits/1dd2d353382978d394919cb278b9941510142d66 is acceptable.

Reply:
I still think it would be better to split the test into two, and validate the return of `_parse_build_system_table`, but I am not gonna block the PR because of that.

Reply:
Done in https://github.com/pypa/build/pull/365/commits/8bab23845d9f002977324a8b928dc1f97fd71c70.